app/controllers/sensor_readings_controller.rb

```diff
@@ -48,8 +48,51 @@ def fake
     end
   end
 
+  def add
+    if manual_params[:sensor_id].nil? || manual_params[:report_id].nil?
+      respond_to do |format|
+        format.html { redirect_to  root_path }
+        format.js  { render :js => "window.location.href='"+root_path+"'"}
+      end
+      return
+    end
+
+    if manual_params[:timestamp].nil? || manual_params[:calibrated_value].nil? || manual_params[:uncalibrated_value].nil?
```

Comment:
If this is supposed to be a validation, it belongs into the `app/sensor/reading.rb` model. Have a look into the Rails guides: http://guides.rubyonrails.org/active_record_validations.html